File: apps/web/src/types/company.ts
/**
 * 会社プロフィールの型定義
 */
export interface Company {
  /** 会社ID */
  id: string;
  /** ユーザーID */
  user_id: string;
  /** 会社名 */
  name: string;
  /** 会社説明 */
  description?: string;
  /** 対応可能地域 */
  regions: string[];
  /** 希望案件規模 - 最小金額 */
  budget_min?: number;
  /** 希望案件規模 - 最大金額 */
  budget_max?: number;
  /** 保有スキル */
  skills: string[];
  /** NGキーワード */
  ng_keywords: string[];
  /** 作成日時 */
  created_at: string;
  /** 更新日時 */
  updated_at: string;
}

/**
 * 会社プロフィール作成リクエスト
 */
export interface CreateCompanyRequest {
  /** 会社名 */
  name: string;
  /** 会社説明 */
  description?: string;
  /** 対応可能地域 */
  regions: string[];
  /** 希望案件規模 - 最小金額 */
  budget_min?: number;
  /** 希望案件規模 - 最大金額 */
  budget_max?: number;
  /** 保有スキル */
  skills: string[];
  /** NGキーワード */
  ng_keywords: string[];
}

/**
 * 会社プロフィール更新リクエスト
 */
export interface UpdateCompanyRequest extends CreateCompanyRequest {}

File: apps/api/schemas/company.py
"""
会社プロフィール関連のPydanticスキーマ

リクエスト/レスポンスのバリデーションとシリアライゼーションを定義します。
"""
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict


class CompanyBase(BaseModel):
    """会社情報の基本スキーマ"""

    name: str = Field(..., min_length=1, max_length=255, description="会社名")
    description: str | None = Field(None, description="会社の説明")
    regions: list[str] = Field(default_factory=list, description="対応可能な都道府県コード配列")
    budget_min: int | None = Field(None, ge=0, description="最小予算（円）")
    budget_max: int | None = Field(None, ge=0, description="最大予算（円）")
    skills: list[str] = Field(default_factory=list, description="保有スキル配列")
    ng_keywords: list[str] = Field(default_factory=list, description="NGキーワード配列")


class CompanyCreate(CompanyBase):
    """会社作成用スキーマ"""

    pass


class CompanyUpdate(BaseModel):
    """会社更新用スキーマ（すべてのフィールドがOptional）"""

    name: str | None = Field(None, min_length=1, max_length=255, description="会社名")
    description: str | None = Field(None, description="会社の説明")
    regions: list[str] | None = Field(None, description="対応可能な都道府県コード配列")
    budget_min: int | None = Field(None, ge=0, description="最小予算（円）")
    budget_max: int | None = Field(None, ge=0, description="最大予算（円）")
    skills: list[str] | None = Field(None, description="保有スキル配列")
    ng_keywords: list[str] | None = Field(None, description="NGキーワード配列")


class CompanyResponse(CompanyBase):
    """会社情報レスポンス用スキーマ"""

    id: str = Field(..., description="会社ID (UUID)")
    user_id: str = Field(..., description="ユーザーID (UUID)")
    created_at: datetime = Field(..., description="作成日時")
    updated_at: datetime = Field(..., description="更新日時")

    model_config = ConfigDict(from_attributes=True)
